docs/spark/how-to-guides/streaming.md

```diff
@@ -0,0 +1,123 @@
+
+To start a new connection, open a command prompt. If you are using Linux, run ```nc -lk 9999``` to connect to localhost on port 9999.
+
+If you are using Windows, run ```nc -vvv -l -p 9999``` to connect to localhost port 9999.
+
+Your Spark program listens for the input you type into this command prompt.
+
+### Connect to a stream with ReadStream()
+
+The `ReadStream()` method returns a `DataStreamReader` that can be used to read streaming data in as a `DataFrame`. Include the host and port information to tell your Spark app where to expect its streaming data.
+
+```CSharp
+DataFrame lines = spark
+    .ReadStream()
+    .Format("socket")
+    .Option("host", hostname)
+    .Option("port", port)
+    .Load();
+```
+
+## Register a user-defined function
+
+A UDF is a *user-defined function.* You can use UDFs in Spark applications to perform calculations and analysis on your data.
+
+```CSharp
+Func<Column, Column> udfArray =
+    Udf<string, string[]>((str) => new string[] { str, $"{str} {str.Length}");
+```
+
+In the above code snippet from [StructuredNetworkCharacterCount.cs](StructuredNetworkCharacterCount.cs), you register a UDF called `udfArray`. This UDF processes each string it receives from the netcat terminal to produce an array that includes: the original string (contained in *str*), the original string concatenated with the length of that original string. 
+
+For example, entering *Hello world* in the terminal would produce an array where:
+
+* array[0] = Hello world
+* array[1] = Hello world 11
+
+## Use SparkSQL
+
+Use SparkSQL to perform various functions on the data stored in your DataFrame. It's common to combine UDFs and SparkSQL to apply a UDF to each row of your DataFrame.
+
+```CSharp
+DataFrame arrayDF = lines.Select(Explode(udfArray(lines["value"])));
+```
+
+In the above code snippet from [StructuredNetworkCharacterCount.cs](StructuredNetworkCharacterCount.cs), *udfArray* is applied to each value in your DataFrame (which represents each string read in from our netcat terminal). Apply the SparkSQL method `Explode` to put each entry of your array in its own row. Finally, use `Select` to place the columns you've produced in the new DataFrame *arrayDF.*
+
+## Display your stream
+
+Use `DataFrame.WriteStream()` to establish characteristics of your output, such as printing results to the console, displaying only the most recent output.
+
+```CSharp
+StreamingQuery query = arrayDf
+    .WriteStream()
+    .Format("console")
+    .Start();
+```
+
+## Run your code
+
+Structured streaming in Spark processes data through a series of small **batches.** 
+When you run your program, the command prompt where you establish the netcat allows you to start typing.
+
+When you hit *enter* after entering data in the command prompt, Spark considers that a batch and runs the UDF. 
+
+### Run on Windows
+
+After starting a new netcat session, open a new terminal and run your `spark-submit` command, similar to the following command:
```

Comment:
```suggestion
After starting a new netcat session, open a new terminal and run your `spark-submit` command, similar to the following Windows command:
```

Reply:
This is confusing a little bit for me. You're changing the title because:

> Users can run on any platform (not just Windows)

but adding Windows to this sentence.

Reply:
@Youssef1313 Users can run .NET for Spark apps on any platform (Windows, Linux, Mac), so I didn't want new users to see this header and think, "Oh so I have to run this app on Windows?" 

It's just that the `spark-submit` code example here is specifically a Windows command (to give users an example of a command they could run to run their code, but it's not their only option since they could've for instance run with Linux). 

Does that shed any light? Definitely open to feedback/other options.

Reply:
@bamurtaugh , Thanks for clarification. If this command is windows-specific, then I think it's good idea to provide similar commands that are mac- and linux- specific.

Reply:
@Youssef1313 that makes sense to me. After investigating further, it appears the Windows and Linux commands should be the same (aside from the exact syntax for file paths). We could update this line/section to something like the following:

-After starting a new netcat session, open a new terminal to build and run your app
-Build your app: `dotnet build`
-Run your app via `spark-submit`: `spark-submit --class org.apache.spark.deploy.dotnet.DotnetRunner --master local /<path>/to/microsoft-spark-<version>.jar dotnet /<path>/to/netcoreapp<version>/StructuredNetworkCharacterCount.dll localhost 9999`
-For more instructions about running your .NET for Spark app, check out the guides for [building on Windows](https://github.com/dotnet/spark/blob/master/docs/building/windows-instructions.md#using-net-core-cli-for-net-core-21x) or [Building on Ubuntu](https://github.com/dotnet/spark/blob/master/docs/building/ubuntu-instructions.md).